src/build.d

```diff
@@ -488,6 +497,46 @@ alias style = makeDep!((builder, dep)
         ]);
 });
 
+alias man = makeDep!((builder, dep) {
+    alias genMan = methodInit!(Dependency, (genManBuilder, genManDep) => genManBuilder
+        .target(env["G"].buildPath("gen_man"))
+        .sources([
+            dmdRepo.buildPath("docs/gen_man.d"),
+            env["D"].buildPath("cli.d")])
+        .command([
+            env["HOST_DMD_RUN"],
+            "-I" ~ srcDir,
+            "-of" ~ genManDep.target]
+            ~ flags["DFLAGS"]
+            ~ genManDep.sources)
+        .msg(genManDep.command.join(" "))
+    );
+
+    const genManDir = env["GENERATED"].buildPath("docs/man");
+    alias dmdMan = methodInit!(Dependency, (dmdManBuilder, dmdManDep) => dmdManBuilder
+        .target(genManDir.buildPath("man1/dmd.1"))
+        .deps([genMan, mkdirDep(dmdManDep.target.dirName)])
+        .commandFunction(() {
+            if (updateIfChanged(dmdManDep.target, genMan.target.execute.output))
+                writefln("Generated '%s'", dmdManDep.target);
+            else writefln("'%s' already up-to-date", dmdManDep.target);
+        })
+    );
+    builder
+    .name("man")
+    .description("generate and prepare man files")
+    .deps([dmdMan].chain(
+        "man1/dumpobj.1 man1/obj2asm.1 man5/dmd.conf.5".split
+        .map!(e => methodInit!(Dependency, (manFileBuilder, manFileDep) => manFileBuilder
+            .target(genManDir.buildPath(e))
+            .sources([dmdRepo.buildPath("docs/man", e)])
+            .deps([mkdirDep(manFileDep.target.dirName)])
+            .command(["cp", manFileDep.sources[0], manFileDep.target])
+            .msg(manFileDep.command.join(" "))
+        ))
```

Comment:
Probably better as a local function/lambda before `man`. That would clearly seperate these dependencies and will probably be easier when using `buildPath`

Reply:
(Comment is referring to the entire map-lambda. Apparently GitHub does not handle multiline review comments - or i messed up)